src/sage/misc/sagedoc.py

```diff
@@ -513,7 +513,21 @@ def process_dollars(s):
     'oeis': ('https://oeis.org/%s', 'OEIS sequence %s'),
     'doi': ('https://doi.org/%s', 'doi:%s'),
     'pari': ('https://pari.math.u-bordeaux.fr/dochtml/help/%s', 'pari:%s'),
-    'mathscinet': ('https://www.ams.org/mathscinet-getitem?mr=%s', 'MathSciNet %s')
+    'mathscinet': ('https://www.ams.org/mathscinet-getitem?mr=%s', 'MathSciNet %s'),
+    'common_lisp': ('https://www.lispworks.com/documentation/lw50/CLHS/Body/%s.htm', 'Common Lisp: %s'),
+    'ecl': ('https://ecl.common-lisp.dev/static/manual/%s.html', 'ECL: %s'),
+    'gap': ('https://gap-system.org/Manuals/doc/ref/%s_mj.html', 'GAP: %s'),
+    'gap_package': ('https://www.gap-system.org/Packages/%s.html', 'GAP package %s'),
```

Comment:
Just as heads-up, those URLs are probably not going to exist for much longer. Not sure what the goal here is, is it to link to the homepages of these packages? Or their documentation? Or what?

Reply:
Anything really, I don't need anything very specific here for now. Package documentation would be fine.

Reply:
@fingolfin What URL would you recommend to use for the package documentation?

Reply:
@fingolfin And more generally (not for this PR) -- what would be the best way to link to specific functions or other objects in the GAP documentation in a stable way? 
(For packages that use Sphinx to build their documentation, we use the cross-referencing tool Intersphinx for this purpose.)

Reply:
While we still mirror the docs of distributed GAP packages on www.gap-system.org, the URLs there are not quite uniform, as the entry pages have different names / paths for each package.

This information is contained in the metadata of each package. That metadata is available in a JSON file (e.g. <https://github.com/gap-system/gap/releases/download/v4.13.0/package-infos.json.gz> for the packages distributed with GAP 4.13.0). I can provide further details if desired.

On the website we only link to the manual for the latest GAP release, and the manuals of the packages within that. So these are not really "stable".

I hope that we can eventually enhance this by adding version pickers. In preparation for this future (which may never come *sigh* but I am hoping), I made sure that from a certain point on, we had separate URLs for each GAP version -- these are not currently used anywhere! But as I wrote, I hope this will change.

The "regular" URLs for our mirrors of package manuals look like these (note the different paths after the package name)
- <https://docs.gap-system.org/pkg/smallsemi/doc/chap0.html>
- <https://docs.gap-system.org/pkg/transgrp/htm/chapters.htm>

The "versioned" ones look like this:
- <https://docs.gap-system.org/v4.13.0/pkg/smallsemi/doc/chap0.html>
- <https://docs.gap-system.org/v4.13.0/pkg/transgrp/htm/chapters.htm>

(So they are versioned by GAP version, not by package version. We could debate this at length, but let's not).

Anyway, so my suggestion would be to use these URLs.

But the annoying bit remains that you can't know the full URLs (`.../doc/chap0.html` or `.../doc/chap0_mj.html` or `..../htm/chapters.htm` or maybe something else) without consulting that JSON file.

But perhaps we could just add a step to [the script which updates the manuals on the GAP website for a new release](https://github.com/gap-system/GapWWW/blob/master/etc/extract_manuals.py), and add redirects ...
- from <https://docs.gap-system.org/v4.13.0/pkg/smallsemi/> to <https://docs.gap-system.org/v4.13.0/pkg/smallsemi/doc/chap0.html>,
- from <https://docs.gap-system.org/v4.13.0/pkg/transgrp/> to <https://docs.gap-system.org/v4.13.0/pkg/transgrp/htm/chapters.htm>,
- and so on.

Since [that script](https://github.com/gap-system/GapWWW/blob/master/etc/extract_manuals.py) already is parsing the JSON data and more, it should be fairly easy to do. We use Apache, so the script could just produce a `.htaccess` file with suitable rules... (Actually it might be even easier to write a separate script which consumes a `package-infos.json` and spits out the `.htaccess` file; then it'd be very easy to apply it retroactively to prior GAP releases).

All that said, I am very busy with other stuff right now so I don't want to promise anything. If someone would like to help out, I can provide further details.

Reply:
Thanks a lot for these details!

> they are versioned by GAP version, not by package version.

That's good enough for us here, I think, thanks.

> add redirects ... from https://docs.gap-system.org/v4.13.0/pkg/smallsemi/ to https://docs.gap-system.org/v4.13.0/pkg/smallsemi/doc/chap0.html,

That could be nice.

More important for use would be a way to link to specific functions in a stable (or computable) way though. For example, https://docs.gap-system.org/doc/ref/chap16.html#X821888E77EB43F67 leads to Permanent, but can I assume that this is a permalink?

There's no urgency for either of these things. Here on this PR I'm really just making a first approximation with the goal of normalizing/encouraging creating richer documentation links to our upstream projects. We can revisit this when you have a bit more time!

Reply:
@fingolfin For example, might there be interest on your side in extending GAPDoc so it can output RST files and then go through Sphinx for the HTML build?

Reply:
Those anchors are hashes and are essentially permanent. For our website we basically keep an index file (see https://github.com/gap-system/GapWWW/blob/master/_data/help.json which is exported from GAP via [this script](https://github.com/gap-system/gap/blob/master/dev/releases/HelpLinks-to-JSON.g).) 

Regarding "output RST files and then go through Sphinx for the HTML build": I am not really familiar with RST files and Sphinx. I am pretty sure I'd have a hard time convincing most GAP package maintainers to learn about those and installing relevant parts. But I guess it might still be potentially useful as an optional backend with added features (search, the ability to keep multiple versions and switch between them come to mind) that perhaps then is only used by the GAP website (and possibly via GH Action jobs also on package websites, if we can provide package authors with a "drop in and forget" solution). 

As such, I would find that potentially interesting. To be fully useful, also the GAP help system would have to be taught to deal with the resulting HTML (GAP can be set up so that the REPL help opens the HTML docs at the correct position). I guess this is less important if it only is about producing online versions of those docs.

I have zero interest in working on that, but if someone else did, I'd be happy to try it out and give feedback.

I have no idea if there is any chance to get such a patch into GAPDoc, though (the maintainer sometimes has... strong oppinions). But it might not even be necessary, GAPDoc already has three "backends" ("plain text with escape sequences" for the REPL help; HTML; PDF-via-latex), and I think this is done in a way that makes it easy to add extra backends. So such a thing could perhaps be done via an extension package.

Reply:
From the GAPDoc:
```
  # - add hash strings of the latter for links in HTML and PDF version
  #   (such that links remain stable as long as the (sub)section exists
  #   and stays in the same chapter)
```

Reply:
> I guess it might still be potentially useful as an optional backend with added features (search, the ability to keep multiple versions and switch between them come to mind) that perhaps then is only used by the GAP website (and possibly via GH Action jobs also on package websites, if we can provide package authors with a "drop in and forget" solution).

Yes, that's what I would imagine. No need to bring local building by GAP users/developers into the equation.

> I have zero interest in working on that

Likewise, but that's a kind of project that would pretty easy to hire someone for. No math required!

Reply:
Ah, I wish I could just hire someone for that kind of work, but in general this is very difficult with German university funding and general regulations on hiring people :/. Essentially I can hire local students (but good luck finding one with the skills). *sigh*